nano/lib/tomlconfig.cpp

```diff
@@ -186,6 +186,36 @@ void nano::tomlconfig::erase_default_values (tomlconfig & defaults_a)
 	erase_defaults (defaults_l.get_tree (), self.get_tree (), get_tree ());
 }
 
+// Merges two TOML configurations and commenting values that are identical
+std::string nano::tomlconfig::merge_defaults (nano::tomlconfig & current_config, nano::tomlconfig & default_config)
+{
+	// Serialize both configs to commented strings
+	std::string defaults_str = default_config.to_string (true);
+	std::string current_str = current_config.to_string (true);
+
+	// Read both configs line by line
+	std::istringstream stream_defaults (defaults_str);
+	std::istringstream stream_current (current_str);
+	std::string line_defaults, line_current, result;
+
+	while (std::getline (stream_defaults, line_defaults) && std::getline (stream_current, line_current))
+	{
+		if (line_defaults == line_current)
+		{
+			// Use default value
+			result += line_defaults + "\n";
+		}
+		else
+		{
+			// Non default value. Removing the # to uncomment
```

Comment:
It is always commented because the tostring() function takes a boolean value that determines if all parameters should be commented or not:
```
	// Serialize both configs to commented strings
	std::string defaults_str = default_config.to_string (true);
	std::string current_str = current_config.to_string (true);
```

Reply:
It is also assuming that there are characters after the '#' character.
How about you put an assert to check ensure that pop will never be npos and an assert to check that the string length is larger than pos.
To at least make the assumptions explicit.